src/formatting/items.rs

```diff
@@ -1816,7 +1816,7 @@ pub(crate) fn rewrite_struct_field(
 
     let is_prefix_empty = prefix.is_empty();
     // We must use multiline. We are going to put attributes and a field on different lines.
-    let field_str = rewrite_assign_rhs(context, prefix, &*field.ty, shape)?;
+    let field_str = rewrite_assign_rhs(context, prefix.trim_end(), &*field.ty, shape)?;
     // Remove a leading white-space from `rewrite_assign_rhs()` when rewriting a tuple struct.
     let field_str = if is_prefix_empty {
         field_str.trim_start()
```

Comment:
I'm a little worried about unintentional side effects here. I think the real issue is that there's an assumption in this function that tuple struct fields will have no prefix (visibility, etc.) which obviously doesn't always pan out. I'd rather just add an additional branch between this current if and else which checks the end and beginning of the prefix and initial field_str respectively, and if both contain a space then just trim one of them.

what do you think?

Reply:
Actually, I was wrong about this sorry. The real triggering scenario is an ident-less struct field which has some public/crate visibility and the formatted type cannot fit on the same line :upside_down_face: 

As such we do indeed want to drop any trailing space off the end, since the util function used isn't expecting the lhs to have one, good catch!